snuba/migrations/backfill_errors.py

```diff
@@ -6,462 +6,169 @@
-        event.sdk_name,
-        event.sdk_version,
-        event.http_method,
-        event.http_referer,
-        event.tags_key,
-        event.tags_value,
-        event.contexts_key,
-        event.contexts_value,
-        event.transaction_name or "",
-        span_id,
-        trace_id,
-        event.partition or 0,
-        event.offset or 0,
-        event.message_timestamp,
-        event.retention_days,
-        event.deleted,
-        event.group_id,
-        UUID(event.primary_hash),
-        event.received or 0,
-        event.message or "",
-        event.title or "",
-        event.culprit or "",
-        event.level,
-        event.location,
-        event.version,
-        event.type or "",
-        event.exception_stacks_type,
-        event.exception_stacks_value,
-        event.exception_stacks_mechanism_type,
-        event.exception_stacks_mechanism_handled,
-        event.exception_frames_abs_path,
-        event.exception_frames_colno,
-        event.exception_frames_filename,
-        event.exception_frames_function,
-        event.exception_frames_lineno,
-        event.exception_frames_in_app,
-        event.exception_frames_package,
-        event.exception_frames_module,
-        event.exception_frames_stack_level,
-        event.sdk_integrations,
-        event.modules_name,
-        event.modules_version,
-    )
-
-    return {
-        name: value
-        for name, value in zip(Error.field_names(), error._asdict().values())
-    }
-
-
-class MigratedEvent(NamedTuple):
-    event_id: str
-    project_id: int
-    timestamp: datetime
-
-    def __repr__(self) -> str:
-        return f"{self.event_id} {self.project_id} {self.timestamp}"
+from snuba.migrations.operations import InsertIntoSelect
+
+
+def format_date(day: date) -> str:
+    return day.strftime("%Y-%m-%d")
+
+
+def get_monday(day: date) -> date:
+    return day - timedelta(days=day.weekday())
+
+
+WINDOW = timedelta(days=7)
+BEGINNING_OF_TIME = get_monday((datetime.utcnow() - timedelta(days=90)).date())
+
+
+COLUMNS = [
+    ("project_id", "project_id"),
+    ("timestamp", "timestamp"),
+    (
+        "event_id",
+        "toUUID(UUIDNumToString(toFixedString(unhex(toString(event_id)), 16)))",
+    ),
+    ("platform", "ifNull(`platform`, '')"),
+    ("environment", "environment"),
+    ("release", "`sentry:release`"),
+    ("dist", "`sentry:dist`"),
+    ("ip_address_v4", "if(position(ip_address, '.') > 0, toIPv4(`ip_address`), Null)"),
+    ("ip_address_v6", "if(position(ip_address, ':') > 0, toIPv6(`ip_address`), Null)"),
+    ("user", "ifNull(`sentry:user`, '')"),
+    ("user_id", "user_id"),
+    ("user_name", "username"),
+    ("user_email", "email"),
+    ("sdk_name", "sdk_name"),
+    ("sdk_version", "sdk_version"),
+    ("http_method", "http_method"),
+    ("http_referer", "http_referer"),
+    ("tags.key", "tags.key"),
+    ("tags.value", "tags.value"),
+    ("contexts.key", "contexts.key"),
+    ("contexts.value", "contexts.value"),
```

Comment:
These are going to miss some values:
https://github.com/getsentry/snuba/blob/master/snuba/datasets/events_processor.py#L97-L98

When we promote a context (not so for the tags) we remove them from the conrtext map, so the contexts array from the old events table need to be amended with the promoted contexts keys and values. 
I thought this was the blocker for using the INSERT INTO.